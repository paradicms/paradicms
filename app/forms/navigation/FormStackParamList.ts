import {
  FormDataSummary,
  FormModelSummary,
  FormShapeSummary,
} from "@paradicms/shacl-forms";

export type FormStackParamList = {
  FormErrorViewerScreen: {
    formData: FormDataSummary | null;
    formNodeData: FormModelSummary | null;
    formNodeTypeData: FormModelSummary | null;
    error: any;
  };
  FormNodeDataEditorScreen: {
    formData: FormDataSummary;
    formNodeData: FormModelSummary;
    formNodeTypeData: FormModelSummary;
  };
  FormNodeTypeDataViewerScreen: {
    formData: FormDataSummary;
    formNodeTypeData: FormModelSummary;
  };
  FormNodeDataViewerScreen: {
    formData: FormDataSummary;
    formNodeData: FormModelSummary;
    formNodeTypeData: FormModelSummary;
  };
  FormDataSummariesViewerScreen: {formShape: FormShapeSummary};
  FormDataViewerScreen: {formData: FormDataSummary};
  FormShapeSummariesViewerScreen: undefined;
};
