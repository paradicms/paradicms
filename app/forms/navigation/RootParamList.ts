export type RootParamList = {
  Settings: undefined;
};
