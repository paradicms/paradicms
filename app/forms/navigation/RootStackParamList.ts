import {NavigatorScreenParams} from "@react-navigation/native";
import {RootTabParamList} from "./RootTabParamList";

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type RootStackParamList = {
  Root: NavigatorScreenParams<RootTabParamList> | undefined;
  Modal: undefined;
  NotFound: undefined;
};
