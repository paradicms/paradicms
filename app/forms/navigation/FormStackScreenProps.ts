import {NativeStackScreenProps} from "@react-navigation/native-stack";
import {FormStackParamList} from "./FormStackParamList";

export type FormStackScreenProps<
  Screen extends keyof FormStackParamList
> = NativeStackScreenProps<FormStackParamList, Screen>;
