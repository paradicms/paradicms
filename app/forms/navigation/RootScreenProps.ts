import {DrawerScreenProps} from "@react-navigation/drawer";
import {RootParamList} from "./RootParamList";

export type RootScreenProps<
  Screen extends keyof RootParamList
> = DrawerScreenProps<RootParamList, Screen>;
