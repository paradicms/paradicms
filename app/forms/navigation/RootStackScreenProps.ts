import {NativeStackScreenProps} from "@react-navigation/native-stack";
import {RootStackParamList} from "./RootStackParamList";

export type RootStackScreenProps<
  Screen extends keyof RootStackParamList
> = NativeStackScreenProps<RootStackParamList, Screen>;
