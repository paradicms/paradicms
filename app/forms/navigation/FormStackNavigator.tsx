import * as React from "react";

import {FormDataViewerScreen} from "../screens/FormDataViewerScreen";
import {FormStackParamList} from "./FormStackParamList";
import {FormStackScreenProps} from "./FormStackScreenProps";
import {createNativeStackNavigator} from "@react-navigation/native-stack";
import {FormNodeDataViewerScreen} from "../screens/FormNodeDataViewerScreen";
import {FormNodeTypeDataViewerScreen} from "../screens/FormNodeTypeDataViewerScreen";
import {FormNodeDataEditorScreen} from "../screens/FormNodeDataEditorScreen";
import {FontAwesome} from "@expo/vector-icons";
import {useTheme} from "@rneui/themed";
import {FormErrorViewerScreen} from "../screens/FormErrorViewerScreen";
import {FormShapeSummariesViewerScreen} from "../screens/FormShapeSummariesViewerScreen";
import {FormDataSummariesViewerScreen} from "../screens/FormDataSummariesViewerScreen";

const Stack = createNativeStackNavigator<FormStackParamList>();

export const FormStackNavigator: React.FunctionComponent = () => {
  const {theme} = useTheme();

  return (
    <Stack.Navigator initialRouteName="FormShapeSummariesViewerScreen">
      <Stack.Group>
        <Stack.Screen
          name="FormDataSummariesViewerScreen"
          component={FormDataSummariesViewerScreen}
          options={({
            route,
          }: FormStackScreenProps<"FormDataSummariesViewerScreen">) => ({
            title: route.params.formShape.label,
          })}
        />
        <Stack.Screen
          name="FormDataViewerScreen"
          component={FormDataViewerScreen}
          options={({route}: FormStackScreenProps<"FormDataViewerScreen">) => ({
            title: route.params.formData.label,
          })}
        />
        <Stack.Screen
          name="FormNodeDataEditorScreen"
          component={FormNodeDataEditorScreen}
          options={({
            navigation,
            route,
          }: FormStackScreenProps<"FormNodeDataEditorScreen">) => {
            return {
              headerRight: () => (
                <FontAwesome.Button
                  backgroundColor="transparent"
                  color={theme.colors.primary}
                  name="save"
                  onPress={() =>
                    navigation.push("FormNodeDataViewerScreen", route.params)
                  }
                />
              ),
              title: route.params.formNodeData.label,
            };
          }}
        />
        <Stack.Screen
          name="FormNodeTypeDataViewerScreen"
          component={FormNodeTypeDataViewerScreen}
          options={({
            route,
          }: FormStackScreenProps<"FormNodeTypeDataViewerScreen">) => {
            return {
              title: route.params.formNodeTypeData.label,
            };
          }}
        />
        <Stack.Screen
          name="FormNodeDataViewerScreen"
          component={FormNodeDataViewerScreen}
          options={({
            navigation,
            route,
          }: FormStackScreenProps<"FormNodeDataViewerScreen">) => {
            return {
              headerRight: () => (
                <FontAwesome.Button
                  backgroundColor="transparent"
                  color={theme.colors.primary}
                  name="edit"
                  onPress={() =>
                    navigation.push("FormNodeDataEditorScreen", route.params)
                  }
                />
              ),
              title: route.params.formNodeData.label,
            };
          }}
        />
        <Stack.Screen
          name="FormShapeSummariesViewerScreen"
          component={FormShapeSummariesViewerScreen}
          options={{title: "Shapes"}}
        />
      </Stack.Group>
      <Stack.Group screenOptions={{presentation: "modal"}}>
        <Stack.Screen
          component={FormErrorViewerScreen}
          name="FormErrorViewerScreen"
        ></Stack.Screen>
      </Stack.Group>
    </Stack.Navigator>
  );
};
