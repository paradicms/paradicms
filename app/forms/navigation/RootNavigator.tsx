import * as React from "react";
import {createDrawerNavigator} from "@react-navigation/drawer";
import {FormStackNavigator} from "./FormStackNavigator";

const Drawer = createDrawerNavigator();

export const RootNavigator: React.FunctionComponent = () => {
  return (
    <Drawer.Navigator>
      <Drawer.Screen
        name="Form"
        component={FormStackNavigator}
        options={{headerShown: false}}
      />
      {/*<Drawer.Screen*/}
      {/*  name="Settings"*/}
      {/*  component={SettingsScreen}*/}
      {/*  options={{*/}
      {/*    title: "Settings",*/}
      {/*  }}*/}
      {/*/>*/}
    </Drawer.Navigator>
  );
};
