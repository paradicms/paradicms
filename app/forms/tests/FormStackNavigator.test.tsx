import * as React from "react";
import * as renderer from "react-test-renderer";
import {FormStackNavigator} from "../navigation/FormStackNavigator";
import {NavigationContainer} from "@react-navigation/native";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <NavigationContainer>
        <FormStackNavigator />
      </NavigationContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
