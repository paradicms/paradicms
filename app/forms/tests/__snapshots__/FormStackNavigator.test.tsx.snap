// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Shapes"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        />
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
